app/controllers/income/UpdateIncomeNextYearController.scala

```diff
@@ -85,15 +94,22 @@ class UpdateIncomeNextYearController @Inject()(
 
         DuplicateSubmissionWarningForm.createForm.bindFromRequest.fold(
           formWithErrors => {
-            updateNextYearsIncomeService.get(employmentId, nino) flatMap { model =>
-              val vm = if (model.isPension) {
-                DuplicateSubmissionCYPlus1PensionViewModel(model.employmentName, model.newValue.get)
-              } else {
-                DuplicateSubmissionCYPlus1EmploymentViewModel(model.employmentName, model.newValue.get)
-              }
 
-              Future.successful(
-                BadRequest(views.html.incomes.nextYear.updateIncomeCYPlus1Warning(formWithErrors, vm, employmentId)))
+            updateNextYearsIncomeService.getNewAmount(employmentId).flatMap {
+              case Right(newAmount) =>
+                updateNextYearsIncomeService.get(employmentId, nino) flatMap { model =>
+                  val vm = if (model.isPension) {
+                    DuplicateSubmissionCYPlus1PensionViewModel(model.employmentName, newAmount)
+                  } else {
+                    DuplicateSubmissionCYPlus1EmploymentViewModel(model.employmentName, newAmount)
+                  }
+
+                  Future.successful(BadRequest(
+                    views.html.incomes.nextYear.updateIncomeCYPlus1Warning(formWithErrors, vm, employmentId)))
+                }
```

Comment:
With the exception of the use of BadRequest this is a near duplicate of lines 70-84. Any potential to refactor this to deduplicate?

Reply:
Done